### PR TITLE
Move openmetrics module to oss

### PR DESCRIPTION
**Pull request by ChrsMark**
## What does this PR do?
This PR moves `openmetrics` Metricbeat module under oss directory.

### CODE REVIEW

**Comment by jsoriano** [comment]
CI failure seems related.

It'd be nice to add a changelog entry for this.

File: Jenkinsfile
Comment (kuisathaverat) [inline Jenkinsfile:749]: the `getCommonModuleInTheChangeSet(String directory)` search for a pattern in a directory path, fi the pattern does not exist will return ''. I am not sure of the goal of check a folder with the name of the module exists in the current folder, probably will never exist.

cc @v1v

Comment (ChrsMark) [reply Jenkinsfile:749]: Is pattern checked for the current(=new) state or the old one? In this particular case we have removed the `openmetrics` module complete from x-pack but the tester still understands that is has changes and thus it kicks the tests for it which leads to test's failure later on since the module does not actually exists under x-pack.

Comment (v1v) [reply Jenkinsfile:749]: A bit of context:

Unfortunately this is an edge case when moving files between x-pack and oss beats.

https://github.com/elastic/apm-pipeline-library/pull/535 didn't implement a git diff by excluding files that have been deleted/moved.

So somehow it tries to run the module in `x-pack` when that particular `module` was moved to the `oss`.

Therefore, the validation if the module folder exists could be a good workaround.

> I am not sure of the goal of check a folder with the name of the module exists in the current folder, probably will never exist.

if the module name exists in the `pwd` then it should work, unless `module` is a nested folder somewhere else, if so, it's required to consider that particular case.

Comment (v1v) [reply Jenkinsfile:749]: I've just confirmed what Ivan said, and there is something missing in this logic, something like the below snippet might work, though I don't know if `module` is always a nester folder under `<beat_name_folder>` or `x-pack/<beat_name_folder>`

```suggestion
    if(!fileExists("${directory}/module/${module}")) {
      module = ''
    }
```

**Comment by jsoriano** [comment]
Failing test is flaky, opened issue (https://github.com/elastic/beats/issues/26608) and PR to skip (https://github.com/elastic/beats/pull/26609).

**Review by v1v — APPROVED** [review]

**Review by jsoriano — APPROVED** [review]
Once the changelog entry is added, ship it!

**Review by jsoriano — APPROVED** [review]

**Comment by ChrsMark** [comment]
Wondering why mergify didn't open the backport PR for this 🤔

**Comment by v1v** [comment]
> Wondering why mergify didn't open the backport PR for this 🤔

@ChrsMark , backports are automated with mergify as long as the `backport-v7.<number>` is added.

`<number>` can be changing with the upcoming versions and therefore `.mergify.yml` should have those changes in place. 

`backport-v7.15` does not exist in the `.mergify.yml` file. 

There is some automation to include new versions in the `.mergify.yml`, but I'm not sure those PRs were merged by the person who bumped those release versions.

In fact, the PR is open, https://github.com/elastic/beats/pull/26620 so , I'll leave that operation to the release manager in the beats team, either that PR can be merged if possible, or the mergify is cherry-picked and push to the master branch.

**Comment by v1v** [comment]
I've just merged https://github.com/elastic/beats/pull/26620 and created https://github.com/elastic/beats/pull/26641 to automate the auto-merge.

**Comment by ChrsMark** [comment]
Thanks @v1v !

**Pull request merged**
